### DISCUSSION TITLE
Mass calibration

### DISCUSSION THREAD

**Discussion by swillems** [MannLabs/alphatims#157] [Ideas]
This discussion is intended to optimize mass calibration of AlphaTims. As several people are involved, this discussion should be the central point for transparant communication. Currently interest parties:

@swillems 
@asinha-yyz 
@DarylWM

**Comment by DarylWM** [comment]
Dear Sander,

Thank you for starting the conversation here. Perhaps for my own understanding you could share the architectural vision of AlphaTims and how it fits with the end-to-end scope of AlphaPept. For example, it turned out [the issue I raised](https://github.com/MannLabs/alphatims/issues/145#issue-957823692) affecting my pipeline didn't matter much in the end, as my subsequent mass recalibration step tidied up any mass deltas. More generally, what do you see as fitting with the theme of AlphaTims versus what is better suited to a downstream step in AlphaPept (e.g. mass recalibration)?

Best regards,
Daryl.

**Comment by swillems** [comment]
Dear Daryl,

We are still (re)organizing many aspects AlphaPept ecosystem. In general, our intentations are to develop multiple mudules that take care of different steps of MS data analysis. Several of these modules can function as stand-alone tools, but all of them should at the very least be easy plug-ins in AlphaPept itself. This is also true for AlphaTims. Due to the lack of calibration, it is not yet being used in the official release of AlphaPept, even though we internally have some more flexibility to do so.

As far as the scope goes, I envision AlphaTims to act mostly on the raw data without any sample awareness (e.g. database, searching, scoring, differential expression, ...). As such, I think some crude calibrations based on e.g. calibrant sprays or mass defects are definately possible. Out of experience we do know that calibration after a "first-search" often yields the best results, but this would thus not be part of AlphaTims directly in my opinion and indeed fits better with AlphaPept directly. While I realize that your approach is primarily a targeted one after initial identification, I hope some of the ideas might still be applicable to AlphaTims.

Best,
Sander

**Comment by liquidcarbon** [comment]
@DarylWM what is your approach to mass recalibration?
My impression was that there's a polynomial (maybe simply quadratic) function that converts TOF values to mz, and the coefficients are determined using sodium formate injection, which forms a variety of clusters in both positive and negative mode.  Maybe the same thing could be done with internal standards, or with ubiquitous compounds that are in every sample.  Is it possible to recalibrate in other ways?  I'm seeing 20+ mDa systematic mass error more frequently than I'd like...
